packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -104,17 +111,60 @@ export class ProjectService {
         return project;
     }
 
-    async create(user: SessionUser, data: CreateProject): Promise<Project> {
+    async create(user: SessionUser, data: CreateProject): Promise<string> {
         if (user.ability.cannot('create', 'Project')) {
             throw new ForbiddenError();
         }
-        const adapter = await ProjectService.testProjectAdapter(data);
-        const explores = await adapter.compileAllExplores();
-        const projectUuid = await this.projectModel.create(
-            user.organizationUuid,
-            data,
+
+        const jobUuid = await this.startJob(undefined);
+
+        this.createProject(user, data, jobUuid);
+
+        return jobUuid;
+    }
+
+    async createProject(
+        user: SessionUser,
+        data: CreateProject,
+        jobUuid: string,
+    ): Promise<Project> {
+        // Initialize job steps
+        await this.jobModel.createJobSteps(jobUuid, [
+            JobStepType.TESTING_ADAPTOR,
+            JobStepType.COMPILING,
+            JobStepType.CREATING_PROJECT,
+            JobStepType.CACHING,
+        ]);
+        const adapter = await this.jobModel.tryJobStep(
+            jobUuid,
+            JobStepType.TESTING_ADAPTOR,
+            async () => ProjectService.testProjectAdapter(data),
+        );
+        const explores = await this.jobModel.tryJobStep(
+            jobUuid,
+            JobStepType.COMPILING,
+            async () => adapter.compileAllExplores(),
+        );
+
+        const projectUuid = await this.jobModel.tryJobStep(
+            jobUuid,
+            JobStepType.CREATING_PROJECT,
+            async () => this.projectModel.create(
+                    user.organizationUuid,
+                    data,
+                ),
+        );
+
+        await this.jobModel.tryJobStep(
+            jobUuid,
+            JobStepType.CACHING,
+            async () => {
+                await this.projectModel.saveExploresToCache(
+                    projectUuid,
+                    explores,
+                );
+            },
```

Comment:
I think we are missing setting the job status to done at the end of all these steps.